ledger/simulation/trace.go

```diff
@@ -87,6 +88,15 @@ func (eo ResultEvalOverrides) AllowMoreLogging(allow bool) ResultEvalOverrides {
 	return eo
 }
 
+// SetExtraBudget method sets ExtraBudget field in tracer.Result.
+// It omits nil or 0 in budget *uint64 pointer.
+func (eo ResultEvalOverrides) SetExtraBudget(budget *uint64) ResultEvalOverrides {
+	if budget != nil && *budget != 0 {
```

Comment:
It seems like `Request.ExtraBudget` does not need to be a pointer if there's no distinction between 0 and nil.

Reply:
see [request extra budget ptr to instance](https://github.com/algorand/go-algorand/pull/5354/commits/9b305bef9826dec74a92965b34eb204aa2ac6916)